### ISSUE TITLE
text/template: undefined variable with range after Clone

### ISSUE THREAD

**Issue by Eraac** [golang/go#24791]
### What version of Go are you using (`go version`)?
```
go version go1.10.1 darwin/amd64
```

### Does this issue reproduce with the latest release?
commit 08304e886790cbf0ef4bca8158dcb7d99f3a67e7

### What operating system and processor architecture are you using (`go env`)?
```
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/kevin/Library/Caches/go-build"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/kevin/Code/go"
GORACE=""
GOROOT="/usr/local/Cellar/go/1.10.1/libexec"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.10.1/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/43/45xhjfkn0xbbr4flsgzh879h0000gn/T/go-build334100114=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

### What did you do?

In html/template, initialize variable and use it in inner loop

-> full code for reproduce in comment

### What did you expect to see?

Have defined variables (work well with 1.9.5)

### What did you see instead?
```
executing \"flash\" at <\"success\">: undefined variable: $level"}
```

The line throw the error is the first range, not the $level in `<div class>`

**Comment by mvdan** [comment]
Please provide a full working example to reproduce the error. It would also help if you simplified the code - I would imagine that 20 lines of Go code should be enough to showcase any template library bug.

**Comment by Eraac** [comment]
I am sorry, i have few line of code for reproduce the bug
```go
package main

import (
	"html/template"
	"os"
)

type (
	Data struct {
		View   View
		Levels []string
	}

	View struct{}
)

func (v View) Flash(lvl string) []interface{} {
	return []interface{}{"hello"}
}

func main() {
	t, err := template.New("tpl").Parse(`{{ define "layout" }}
    {{ block "flash" . }}
        {{ range $level := .Levels }}
            {{ range $.View.Flash . }}
                <div class="alert alert-{{ $level }}" role="alert">
                    {{ . }} <button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button>
                </div>
            {{ end }}
        {{ end }}
    {{ end }}

    {{ block "content" .}}{{end}}
{{ end }}
`)

	if err != nil {
		panic(err)
	}

	t2, err := t.Clone()

	if err != nil {
		panic(err)
	}

	_, err = t2.Parse(`{{ block "content" . }}my content{{ end }}`)

	if err != nil {
		panic(err)
	}

	err = t2.ExecuteTemplate(os.Stdout, "layout", Data{View: View{}, Levels: []string{"info", "warning", "success"}})

	if err != nil {
		panic(err)
	}
}
```

In one template (no clone line 41), the bug doesn't appears

And the error
`panic: template: tpl:3:27: executing "flash" at <.Levels>: undefined variable: $level`

**Comment by ianlancetaylor** [comment]
Thanks for reporting this.  It will be fixed in the future 1.11 release.
